runtime/lua/vim/shared.lua

```diff
@@ -513,5 +513,111 @@ function vim.is_callable(f)
   return type(m.__call) == 'function'
 end
 
+
+--- Returns an object with methods to be used to log to `filename`
+--
+-- Example:
+--   local log = vim.new_logger('vim-lsp.log')
+--   local _ = log.info() and log.info('Log this message', some_var, another_var)
```

Comment:
You don't HAVE to do the first part of this call though, right?

It can just be:

```lua
log.info("Log this message", some_var, other_var)
```

I don't think we should have the check in for the canonical example since it will confuse people into think they have to do that.

Reply:
Yep, the first part is optional to do a log-level check first.